### PR TITLE
FM-1017 SAR Template Revisions

### PR DESCRIPTION
**Pull request by davidatkinsuk**
Revision of SAR Templates and Data following review:

* Remove any unused fields from SAR data
* Remove duplicate application, assessments etc. as no longer required as not showing draft entities
* Add test data for all domain event types and revise template fields for domain events, ensuring minimum non-duplicated data is provided
* General tidy up of SAR code as required

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/CAS1SubjectAccessRequestRepository.kt [hunk 1]
```diff
@@ -64,8 +59,8 @@ from
         cas_1_application_user_details case_manager on
 		    case_manager.id = apa.case_manager_cas1_application_user_details_id
 	left join 
-        cas_1_application_user_details app_user on
-		    app_user.id = apa.applicant_cas1_application_user_details_id
+        users app_user on
+		    app_user.id = a.created_by_user_id
 	left join 
         users created_by_user on
 		    created_by_user.id = a.created_by_user_id
```
[the remark this answers is not in the document]
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/CAS1SubjectAccessRequestRepository.kt:66]: we'll address this on a separate PR

**Review by williamfalconeruk — APPROVED** [review]
i've had a run through, looks ok from my perspective.
